### PR TITLE
[Merged by Bors] - feat(Topology/Sheaves/Stalks): Generalize to `stalkPushforward_iso_of_inducing`

### PR DESCRIPTION
**Pull request by erdOne**
port of https://github.com/leanprover-community/mathlib3/pull/17539

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by chrisflav — COMMENTED** [review]
Please merge master, to incorporate the `Inducing` -> `IsInducing` rename.

File: Mathlib/Geometry/RingedSpace/OpenImmersion.lean [hunk 1]
```diff
@@ -737,8 +737,8 @@ theorem of_stalk_iso {X Y : SheafedSpace C} (f : X ⟶ Y) (hf : IsOpenEmbedding
       rintro ⟨_, y, hy, rfl⟩
       specialize H y
       delta PresheafedSpace.Hom.stalkMap at H
-      haveI H' :=
-        TopCat.Presheaf.stalkPushforward.stalkPushforward_iso_of_isOpenEmbedding C hf X.presheaf y
+      haveI H' := TopCat.Presheaf.stalkPushforward.stalkPushforward_iso_of_inducing C
+        hf.toInducing X.presheaf y
```
[the remark this answers is not in the document]
Comment (erdOne) [reply Mathlib/Geometry/RingedSpace/OpenImmersion.lean]: Thanks!

**Comment by chrisflav** [comment]
Thanks!

maintainer merge

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**